### PR TITLE
post on python object and class creation

### PR DESCRIPTION
**Pull request by arunppsg**
I have added draft posts on python objects and class creation in python.

### CODE REVIEW

File: content/posts/python-type.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+tags = ["python"]
++++
+
+Everything in Python has a type (even the type `type`!) because everything is an object.
+
+Objects are abstraction of data in Python.
+All data in a Python program is stored as a object.
+
+An object allows instances of it to be created.
+The created instances, which are also objects are destroyed by garbage collection.
```
Comment (tonydavis629) [inline content/posts/python-type.md]: Maybe reword as 'instances can be created from objects.' I'm not sure what you mean by 'instances are destroyed by garbage collection'. Everything in python is an object which _can_ be destroyed by garabge collections if there are no references to it.

File: content/posts/python-type.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+Example:
+
+```python3
+class Foo():
+    def print_self(self):
+         print ('type ': self, 'id ': id(self))
+
+f = Foo()
+print (type(f))  # output: type <class '__main__.Foo'> id 4408722096
+f.print_self()  # output: type <class '__main__.Foo'> id 4408722096
```
Comment (tonydavis629) [inline content/posts/python-type.md:73]: Maybe make a note that the id is the same because it is the same instance, but if it is a different instance, say g = Foo(), it will be a different id

File: content/posts/python-type.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+In case 1, the token `1` is an object of class int.
+`print (type(1))` returns `<class 'int'>`.
+`print (type(type(1))` and it returns `<class 'type'>`.
+Hence, the token `1` is an object of class int.
+All objects have ids.
+The id of token 1 can be seen by `print(id(1))` and it can be inferred that it is same as `id(a_int)`.
+
+In the assignment `a_int = 1`, `a` becomes a reference to the token `1`.
```
Comment (tonydavis629) [inline content/posts/python-type.md]: I think `a` should be `a_int`?

File: content/posts/python-class-creation.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    def __init__(self):
+        pass
+
+if __name__ == "__main__":
+    print ('type of Foo is ', type(Foo))  # shows `<class '__main__.Meta'>`
+    print ('type of Meta is ', type(Meta))  # shows `<class 'type'>`.
+```
+
+The above snippet shows `<class '__main__.Meta'>` and `<class 'type'>`.
+I leave it to a future post for covering customisation of class creation via metaclasses.
```
Comment (tonydavis629) [inline content/posts/python-class-creation.md:104]: I think a simple example would be really interesting. I've never used a meta class.

Comment (arunppsg) [reply content/posts/python-class-creation.md:104]: I thought about it but I wanted to keep this post short. I will do a future post on how to use metaclass.

File: content/posts/python-type.md (hunk 3 above)
Comment (arunppsg) [reply content/posts/python-type.md]: yes, updated it.

File: content/posts/python-type.md (hunk 1 above)
Comment (arunppsg) [reply content/posts/python-type.md]: reworded it, I don't wish to mention `reference counting` as it will take the reader away - I will cover it in a future post.
